backend/src/api/output.py

```diff
@@ -66,5 +68,6 @@ def get_broadcast_type(self, _value: object) -> navi.ExpressionJson | None:
         return None
 
     def enforce(self, value: object) -> object:
-        assert value is not None
+        if not self.allow_none:
+            assert value is not None
```

Comment:
This method is overridden by derived classes, so all derived classes now have a useless `allow_none` property that is not checked.

I would suggest the following: make an `AnyOutput` class where you override `enforce`.